### PR TITLE
Mass change providers

### PR DESCRIPTION
**Pull request by fernandog**
@miigotu can you check this with caution?

@duramato - edit this when you do your tests

Providers working:
RARBG
TD
KAT
TPB
NZBGEEK
MoreThanTV
Speed.cd
TorrentBytes
tntvillage

### CODE REVIEW

**Comment by SiCKRAGEB0T** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your report has all the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure you send a PR against the DEVELOP branch
- [x] Don't send big changes in one go. Split up big PRs into multiple smaller (easier manageble) PRs

Please make sure you also read [contribution guide](https://github.com/SiCKRAGETV/SickRage/blob/master/contributing.md) and followed all the steps.

Thanks!

**Comment by fernandog** [comment]
also some providers don't have seed/leech parsing

**Comment by fernandog** [comment]
https://github.com/SiCKRAGETV/sickrage-issues/issues/2934

File: sickbeard/providers/extratorrent.py [hunk 1]
```diff
@@ -134,12 +134,18 @@ def _doSearch(self, search_strings, search_mode='eponly', epcount=0, age=0, epOb
                         size = int(item['size'])
                         seeders = helpers.tryInt(item['seeders'],0)
                         leechers = helpers.tryInt(item['leechers'],0)
-                        url = item['enclosure']['@url'] if 'enclosure' in item else self._magnet_from_details(item['link'])
+                        download_url = item['enclosure']['@url'] if 'enclosure' in item else self._magnet_from_details(item['link'])
 
-                        if not all([title, url, seeders, seeders >= self.minseed, leechers >= self.minleech, size]):
+                        if not all([title, download_url, seeders, leechers, size, info_hash]):
```
Comment (miigotu) [inline sickbeard/providers/extratorrent.py]: seeders and leechers should not be here, oops.

Comment (fernandog) [inline sickbeard/providers/extratorrent.py]: what if they are None ?

File: sickbeard/providers/freshontv.py [hunk 2]
```diff
@@ -293,12 +293,13 @@ def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj
                                     logger.log(u"Unable to parse torrent leechers content " + self.name + " Traceback: " + traceback.format_exc(), logger.DEBUG)
                                     leechers = 0
 
-                                #Filter unseeded torrent
-                                if mode != 'RSS' and (seeders < self.minseed or leechers < self.minleech):
-                                    logger.log(u"Discarding torrent because it doesn't meet the minimum seeders or leechers: {0} (S:{1} L:{2})".format(title, seeders, leechers), logger.DEBUG)
+                                if not all([title, download_url, id, seeders, leechers]):
```
Comment (miigotu) [inline sickbeard/providers/freshontv.py]: no seeders/leechers here again.

File: sickbeard/providers/hdtorrents.py [hunk 3]
```diff
@@ -204,12 +204,17 @@ def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj
                             # Need size for failed downloads handling
                             if re.match('[0-9]+,?\.?[0-9]* [KkMmGg]+[Bb]+', cells[7].text):
                                 size = self._convertSize(cells[7].text)
-
-                            if not title or not url or not seeders or leechers is None or not size or \
-                                    seeders < self.minseed or leechers < self.minleech:
+                                
+                            if not all([title, download_url, seeders, leechers, size]):
```
Comment (miigotu) [inline sickbeard/providers/hdtorrents.py]: again

File: sickbeard/providers/iptorrents.py [hunk 4]
```diff
@@ -374,25 +376,26 @@ def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj
...
+                                details_url = self.urls['base_url'] + torrent['href']
+                                seeders = int(result.find('td', attrs={'class': 'ac t_seeders'}).string)
+                                id = int(torrent['href'].replace('/details.php?id=', ''))
+                                leechers = int(result.find('td', attrs = {'class' : 'ac t_leechers'}).string)
                             except (AttributeError, TypeError):
                                 continue
 
-                            # Filter unseeded torrent and torrents with no name/url
-                            if mode != 'RSS' and torrent_seeders == 0:
+                            if not all([title, download_url, details_url, seeders, leechers]):
```
Comment (miigotu) [inline sickbeard/providers/iptorrents.py]: omg what was I doing, here too!

File: sickbeard/providers/hdtorrents.py (hunk 3 above)
Comment (fernandog) [inline sickbeard/providers/hdtorrents.py]: all are the same!

File: sickbeard/providers/kat.py [hunk 5]
```diff
@@ -166,16 +166,17 @@ def _doSearch(self, search_strings, search_mode='eponly', epcount=0, age=0, epOb
                         except (AttributeError, TypeError, KeyError):
                             continue
 
-                        # Dont let RSS add items with no seeders either -.-
-                        if not seeders or seeders < self.minseed or leechers < self.minleech:
-                            logger.log(u"Discarding torrent because it doesn't meet the minimum seeders or leechers: {0} (S:{1} L:{2})".format(title, seeders, leechers), logger.DEBUG)
+                        if not all([title, download_url, seeders, leechers, verified, size]):
```
Comment (miigotu) [inline sickbeard/providers/kat.py]: this is why nothing is working....

File: sickbeard/providers/morethantv.py [hunk 6]
```diff
@@ -228,19 +228,16 @@ def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj
                                 continue
 
 
-                            #Filter unseeded torrent
-                            if mode != 'RSS' and (seeders < self.minseed or leechers < self.minleech):
-                                logger.log(u"Discarding torrent because it doesn't meet the minimum seeders or leechers: {0} (S:{1} L:{2})".format(title, seeders, leechers), logger.DEBUG)
+                            if not all([title, download_url, id, seeders, leechers]):
```
Comment (miigotu) [inline sickbeard/providers/morethantv.py]: ...

File: sickbeard/providers/scc.py [hunk 7]
```diff
@@ -212,8 +212,13 @@ def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj
                         except (AttributeError, TypeError):
                             continue
 
-                        if not title or not download_url or seeders < self.minseed or leechers < self.minleech:
-                            logger.log(u"Discarding torrent because it doesn't meet the minimum seeders or leechers: {0} (S:{1} L:{2})".format(title, seeders, leechers), logger.DEBUG)
+                        if not all([title, download_url, id, seeders, leechers]):
```
Comment (miigotu) [inline sickbeard/providers/scc.py]: ...

**Comment by miigotu** [comment]
`if not all([title, download_url, seeders, leechers])`

This is requiring seeders to be > 0 AND leechers > 0

**Comment by fernandog** [comment]
so what's the correct code ?

**Comment by miigotu** [comment]
```python
if not all([title, download_url]):
     continue

if seeders < self.minseed or leechers < self.minleech:
    if mode != 'RSS':
        logger.log(u"Discarding torrent because it doesn't meet the minimum seeders or leechers: {0} (S:{1} L:{2})".format(title, seeders, leechers), logger.DEBUG)
     continue
```

**Comment by miigotu** [comment]
We already know they are not none because of tryInt or catching exceptions.

**Comment by fernandog** [comment]
@miigotu done

**Comment by fernandog** [comment]
@duramato done

**Comment by fernandog** [comment]
@miigotu can you check the second commit?

https://github.com/fernandog/SickRage/commit/70e8e764074a6f0fb6759ac2c7536bd31c1a5fe1

suggestions?
this will reduce number of log lines a lot

**Comment by fernandog** [comment]
also now we only show "Found result..." if not in RSS. reducing more log messages

**Comment by miigotu** [comment]
After this PR is ready, we can go over it really well and then squash.

I have a list of tasks to do for providers.
This is a pretty big job, and this is stage one:

1. All of them should be using the `{'RSS':[],'Episode':[],'Season':[]}` syntax
2. `_get_episode_search_strings` and `_get_season_search_strings` need moved into the `generic.TorrentProvider` or `generic.NZBProvider` (some need modified to move params into `_doSearch()` first, and all removed from individual providers.
3. All should be changed to use the same item indexing `(title, url, seeders, leechers, size, hash)`
4. All should be using params (no %s in urls and no concatenating to urls. url="alwaysthesameurl")

**Comment by fernandog** [comment]
@miigotu can you check&merge this please before I do something wrong with my git? Lol

**Comment by fernandog** [comment]
stopped in iptorrents (the log messages fixing)

**Comment by fernandog** [comment]
done...

**Comment by fernandog** [comment]
when you approve I squash. Or do we keep like that?

File: sickbeard/providers/alpharatio.py [hunk 8]
```diff
@@ -169,8 +169,7 @@ def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj
 
                         #Continue only if one Release is found
                         if len(torrent_rows) < 2:
-                            logger.log(u"The Data returned from " + self.name + " does not contain any torrents",
-                                       logger.DEBUG)
+                            logger.log(u"Data returned from provicer does not contain any torrents", logger.DEBUG)
```
Comment (miigotu) [inline sickbeard/providers/alpharatio.py]: `provider` =P

File: sickbeard/providers/frenchtorrentdb.py [hunk 9]
```diff
@@ -186,17 +186,18 @@ def _doSearch(self, search_strings, search_mode='eponly', epcount=0, age=0, epOb
...
 
                 r = self.getURL(self.url, params=self.search_params)
                 with BS4Parser(r, features=["html5lib", "permissive"]) as html:
                     resultsTable = html.find("div", {"class": "DataGrid"})
-                    logger.log(u"Page opened", logger.DEBUG)
 
                     if resultsTable:
-                        logger.log(u"We have results ", logger.DEBUG)
-                        rows = resultsTable.findAll("ul")
+                          rows = resultsTable.findAll("ul")
```
Comment (miigotu) [inline sickbeard/providers/frenchtorrentdb.py]: bad indent

File: sickbeard/providers/freshontv.py [hunk 10]
```diff
@@ -259,7 +255,7 @@ def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj
 
                             #Continue only if a Release is found
                             if len(torrent_rows) == 0:
-                                logger.log(u"The Data returned from " + self.name + " does not contain any torrent", logger.DEBUG)
+                                logger.log(u"Data returned from provicer does not contain any torrents", logger.DEBUG)
```
Comment (miigotu) [inline sickbeard/providers/freshontv.py]: `provider`

File: sickbeard/providers/hdtorrents.py [hunk 11]
```diff
@@ -164,12 +163,12 @@ def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj
 
             empty = html.find('No torrents here')
             if empty:
-                logger.log(u"Could not find any torrents", logger.ERROR)
+                logger.log(u"Data returned from provicer does not contain any torrents", logger.DEBUG)
```
Comment (miigotu) [inline sickbeard/providers/hdtorrents.py]: `provider`

File: sickbeard/providers/freshontv.py (hunk 10 above)
Comment (fernandog) [inline sickbeard/providers/freshontv.py]: All must be wrong. I copy/past all log message

File: sickbeard/providers/torrentproject.py [hunk 12]
```diff
@@ -123,39 +123,45 @@ def _get_title_and_url(self, item):
...
             size = torrents[i]["torrent_size"]
             trackerUrl = self.urls['api'] + "" + hash + "/trackers_json"
-            logger.log(u'The tracker list is: ' + trackerUrl, logger.DEBUG)
+            #These lines arent normaly needed ,enable only if it fails...
+            #logger.log(u'The tracker list is: ' + trackerUrl, logger.DEBUG)
             jdata = self.getURL(trackerUrl, json=True)
-            magnet = "magnet:?xt=urn:btih:" + hash + "&dn=" + name + "".join(["&tr=" + s for s in jdata])
-            logger.log(u'Magnet URL is: ' + magnet, logger.DEBUG)
-            results.append((name, magnet, size))
+            download_url = "magnet:?xt=urn:btih:" + hash + "&dn=" + name + "".join(["&tr=" + s for s in jdata])
```
Comment (miigotu) [inline sickbeard/providers/torrentproject.py]: name => title

File: sickbeard/providers/torrentproject.py [hunk 13]
```diff
@@ -123,39 +123,45 @@ def _get_title_and_url(self, item):
...
+                
+            #Filter unseeded torrent
+            if seeders < self.minseed or leechers < self.minleech:
+                if mode != 'RSS':
+                    logger.log(u"Discarding torrent because it doesn't meet the minimum seeders or leechers: {0} (S:{1} L:{2})".format(title, seeders, leechers), logger.DEBUG)
+                continue
+
+            if mode != 'RSS':
+                logger.log(u"Found result: %s " % title, logger.DEBUG)
+            results.append((name, download_url, seeders, leechers, size))
```
Comment (miigotu) [inline sickbeard/providers/torrentproject.py]: name => title

File: sickbeard/providers/newznab.py [hunk 14]
```diff
@@ -48,6 +48,8 @@ def __init__(self, name, url, key='0', catIDs='5030,5040', search_mode='eponly',
 
         self.url = self.urls['base_url']
 
+        self.headers.update({'User-Agent': USER_AGENT})
```
Comment (fernandog) [inline sickbeard/providers/newznab.py]: @miigotu about dobnzb, here?

**Comment by fernandog** [comment]
@neoatomic this could break providers images as now we get images using this code

```
    def imageName(self):
        return self.getID() + '.png'
```

some were gifs, and some were caps up

**Comment by miigotu** [comment]
@fernandog getID() lowercases and replaces certain chars. We override that imageName method in providers that have gif or jpg images.

@neoatomic can you convert all of the provider images to png and rename them to the lowercase version of the actual provider's name that is displayed in the logs?

**Comment by fernandog** [comment]
@miigotu no conflicts isn't?

we you are ready, merge. Or if you find any error let me know

File: sickbeard/providers/frenchtorrentdb.py (hunk 9 above)
Comment (miigotu) [inline sickbeard/providers/frenchtorrentdb.py]: @fernandog you gotta fix this

File: sickbeard/providers/torrentproject.py (hunk 12 above)
Comment (miigotu) [inline sickbeard/providers/torrentproject.py]: and this

**Comment by fernandog** [comment]
@neoatomic brokens:

KAT
Strike
SceneAccess
Shazbat.tv
Bluetigers
fnt
animenzb

File: sickbeard/providers/kat.py [hunk 15]
```diff
@@ -166,16 +126,17 @@ def _doSearch(self, search_strings, search_mode='eponly', epcount=0, age=0, epOb
...
-                            logger.log(u"KAT Provider found result " + title + " but that doesn't seem like a verified result so I'm ignoring it", logger.DEBUG)
+                        #Filter unseeded torrent
+                        if seeders < self.minseed or leechers < self.minleech:
+                            if mode != 'RSS':
+                                logger.log(u"Discarding torrent because it doesn't meet the minimum seeders or leechers: {0} (S:{1} L:{2})".format(title, seeders, leechers), logger.DEBUG)
                             continue
 
-                        if not title or not url:
+                        if self.confirmed and not verified and mode != RSS:
+                            logger.log(u"Found result " + title + " but that doesn't seem like a verified result so I'm ignoring it", logger.DEBUG)
```
Comment (fernandog) [inline sickbeard/providers/kat.py]: @miigotu getting this error.

Strange as mode is used a few lines above. Do you know what's wrong?

```
  File "/home/osmc/SickRage/sickbeard/providers/kat.py", line 139, in _doSearch
    if self.confirmed and not verified and mode != RSS:
NameError: global name 'RSS' is not defined

```

Comment (fernandog) [inline sickbeard/providers/kat.py]: @miigotu never mind... so damn...must be 'RSS'

**Comment by fernandog** [comment]
@miigotu strike I'm getting this:

[Strike] :: Requested getURL https://getstrike.net/api/v2/torrents/search/?category=TV&phrase=Dominion S02E13 returned status code is 404: Not Found

{"statuscode":404,"message":"No torrents found."}

should't we handle that in strike.py ?

**Comment by fernandog** [comment]
@miigotu 
this is weird. Strike it's not searching for all exceptions

also IMO we should first search for original name, not exceptions. TBBT should be last.

```
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [ThePirateBay] :: Search string: Big * Theory S09E03
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [ThePirateBay] :: Search string: The Big * Theory S09E03
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [ThePirateBay] :: Search string: TBBT S09E03
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [KickAssTorrents] :: Search string: Big * Theory S09E03
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [KickAssTorrents] :: Search string: The Big * Theory S09E03
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [KickAssTorrents] :: Search string: TBBT S09E03
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [TorrentDay] :: Search string: Big * Theory S09E03 
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [TorrentDay] :: Search string: The Big * Theory S09E03 
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [TorrentDay] :: Search string: TBBT S09E03 
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [Rarbg] :: Search string: Big * Theory S09E03 
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [Rarbg] :: Search string: The Big * Theory S09E03 
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [Rarbg] :: Search string: TBBT S09E03 
DEBUG    SEARCHQUEUE-MANUAL-80379 :: [Strike] :: Search string: TBBT S09E03
```

**Comment by miigotu** [comment]
@fernandog the code looks ok for getstrike.

As long as most of them work with the changes this is a great start.

**Comment by fernandog** [comment]
@miigotu but why strike don't search for all search strings like the others?

**Comment by miigotu** [comment]
we can figure that out then, if everything else is working you did pretty damn good already. Is it mergeable other than that? You tested all? xD

**Comment by duramato** [comment]
@fernandog & @miigotu 
Providers working 
MoreThanTV
Speed.cd
TorrentBytes

**Comment by fernandog** [comment]
Error that need to be fixed:
https://www.irccloud.com/pastebin/w07P2bfT/
https://www.irccloud.com/pastebin/pSqhYJ8X/

**Comment by fernandog** [comment]
squashed

**Pull request merged**